### PR TITLE
Howtoimprovements

### PR DESCRIPTION
**Pull request by mebdivio**

### CODE REVIEW

File: how-to/configure-external-logging.rst [hunk 1]
```diff
@@ -15,8 +15,8 @@ effectively.
 Example using LogDNA
 --------------------
 
-This document will show you how to set up logging using the popular LogDNA service. Using other services the principle
-will be the same, with only some minor differences.
+This document will show you how to set up logging using the popular LogDNA service. Using other services,
+the principle, except for some minor differences, will be the same.
```
Comment (evildmp) [inline how-to/configure-external-logging.rst]: Was better previously, please revert this change.

File: how-to/create-addon.rst [hunk 2]
```diff
@@ -74,7 +74,7 @@ In this case, you will need to add the package to be installed to the
 If your addon *contains* an application
 .......................................
 
-If on the other hand, for example if the application is not available on PyPI,
+On the other hand, if for example, the application is not available on PyPI,
```
Comment (evildmp) [inline how-to/create-addon.rst]: `If on the other hand the application...`

File: how-to/create-addon.rst [hunk 3]
```diff
@@ -83,7 +83,7 @@ simply add it as the inner application directory.
 
     The *inner application directory*, in this case ``susan_example_application``, should have a
     name that matches the *package name* (``susan-example-application``), with underscores
-    substituted for the dashes.
+    substituting for the dashes.
```
Comment (evildmp) [inline how-to/create-addon.rst]: Was better previously, please revert this change

File: how-to/debug-dependency-conflict.rst [hunk 4]
```diff
@@ -47,7 +47,7 @@ same.
 
 ..  admonition:: But I didn't change anything in my project!
 
-    Because of the way pip works, even if you don't change anything at all in your project, simply
+    Because of the way pip works, even if you didn't change anything at all in your project, simply
```
Comment (evildmp) [inline how-to/debug-dependency-conflict.rst]: Was better previously, please revert this change.

File: how-to/install-system-packages.rst [hunk 5]
```diff
@@ -13,9 +13,7 @@ version of Ubuntu Linux.
 
 If your project requires a particular system package, you can include them in
 the Docker image, by listing the commands required to install them - typically,
-using ``RUN apt-get`` - in the ``Dockerfile``.
-
-The commands in the ``Dockerfile`` are executed in order, so an appropriate
+using ``RUN apt-get`` - in the ``Dockerfile``. The commands in the ``Dockerfile`` are executed in order, so an appropriate
```
Comment (evildmp) [inline how-to/install-system-packages.rst]: Was better previously, please revert this change.

File: how-to/install-system-packages.rst [hunk 6]
```diff
@@ -43,4 +41,4 @@ having to rebuild the entire project, jump into a new container running
 
      docker-compose run --rm web bash
 
-This container will disappear (``--rm``) when you exit.
+``--rm`` will make the container disappear when you exit.
```
Comment (evildmp) [inline how-to/install-system-packages.rst]: Was better previously, please revert this change.

File: how-to/interact-database.rst [hunk 7]
```diff
@@ -38,7 +38,7 @@ You can also make the connection manually from within the ``web`` container, for
 
     docker-compose run --rm web psql -h postgres -U postgres db
 
-As well as ``psql`` you can run commands such as ``pg_dump`` and ``pg_restore``. This is useful
+The same as ``psql``, you can also run commands such as ``pg_dump`` and ``pg_restore``. This is useful
 for a number of :ref:`common operations <common-db-operations>`, below.
```
Comment (evildmp) [inline how-to/interact-database.rst]: Was better previously, please revert this change

File: how-to/interact-database.rst [hunk 8]
```diff
@@ -68,7 +68,7 @@ Expose the database's port
 In order to the connect to the database from a tool running directly on your
 own machine, you will need to expose its port (5432).
 
-Add a ports section to the ``db`` service in ``docker-compose.yml`` and map the
+Add a **ports** section to the ``db`` service in ``docker-compose.yml`` and map the
```
Comment (evildmp) [inline how-to/interact-database.rst]: use ``ports``, not **ports**

File: how-to/interact-storage.rst [hunk 9]
```diff
@@ -53,7 +53,7 @@ Interact with your project's Cloud S3 storage
 Obtain your storage access details
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-In the Control Panel for your project, visit the ``/doctor`` URL. For each of the Test and Live
+In the Control Panel of your project, visit the ``/doctor`` URL. For each of the Test and Live
```
Comment (evildmp) [inline how-to/interact-storage.rst]: `In the Dashboard for your project...`

File: how-to/interact-storage.rst [hunk 10]
```diff
@@ -168,7 +168,7 @@ bucket::
            PRE filer_public_thumbnails/
 
 
-Or, to copy (``cp``) a file from your own computer to S3::
+or, to copy (``cp``) a file from your own computer to S3::
```
Comment (evildmp) [inline how-to/interact-storage.rst]: Please revert, original is preferred.

File: how-to/interact-storage.rst [hunk 11]
```diff
@@ -188,8 +188,7 @@ Additional usage information
 
 Run ``aws s3 help`` for more information on commands, or refer to the `AWS CLI Command Reference
 <http://docs.aws.amazon.com/cli/latest/reference/s3/index.html>`_. The AWS CLI can maintain
-multiple profiles and offers other features but it's beyond the scope of this documentation to
-explain that here.
+multiple profiles and offers other features but it's beyond the scope of this documentation.
```
Comment (evildmp) [inline how-to/interact-storage.rst]: Please revert, original is preferred.

File: how-to/live-performance.rst [hunk 12]
```diff
@@ -3,8 +3,8 @@
 How to fine-tune your server's performance
 ==========================================
 
-In the Divio Cloud Control Panel for your project, you can easily change the resources (instances,
-RAM, storage, transfer) allocated to your project to adjust for its needs.
+In the Divio Cloud Control Panel of your project, you can easily change the resources (instances,
```
Comment (evildmp) [inline how-to/live-performance.rst]: `In the Divio Dashboard for your project, you can change...`

File: how-to/live-performance.rst [hunk 13]
```diff
@@ -3,8 +3,8 @@
...
+RAM, storage, transfer) allocated to your project to adjust to its needs.
```
Comment (evildmp) [inline how-to/live-performance.rst]: Please revert, original is preferred.

File: how-to/live-performance.rst [hunk 14]
```diff
@@ -102,7 +102,7 @@ Disable or reconfigure ``UWSGI_CHEAPER`` (uWSGI cheaper mode)
 By default, projects use `uWSGI's cheaper mode <https://uwsgi-docs-additions.readthedocs.io/en/latest/Cheaper.html>`_.
 
 When the site is idling in cheaper mode, uWSGI will dismiss unneeded web workers. This saves RAM, and is the
-recommended configuration for most projects. In some circumstances however it can be advantageous to disable this mode,
+recommended configuration for most projects. In some circumstances however, it can be advantageous to disable this mode,
```
Comment (evildmp) [inline how-to/live-performance.rst]: Please revert, original is preferred.

File: how-to/local-in-live-mode.rst [hunk 15]
```diff
@@ -24,7 +24,7 @@ Build the project in the normal way (``docker-compose build web``) if there have
 Disable the ``volumes`` behaviour in ``docker-compose.yml`` (optional)
 -----------------------------------------------------------------------
 
-If your ``Dockerfile`` includes commands that mean that files in the project are processed at build time (say,
+If your ``Dockerfile`` includes commands it means that files in the project are processed at build time (say,
```
Comment (evildmp) [inline how-to/local-in-live-mode.rst]: Please revert, original is preferred.

File: how-to/local-in-live-mode.rst [hunk 16]
```diff
@@ -34,7 +34,7 @@ In such a case, comment out the line::
 
 in ``docker-compose.yml``. Note that this means that further
 changes you make to the project files on your host system will not be reflected inside the container until the
-line is restored and the project restarted.
+line is restored and the project is restarted.
```
Comment (evildmp) [inline how-to/local-in-live-mode.rst]: Please revert, original is preferred.

File: how-to/password-protect-project.rst [hunk 17]
```diff
@@ -29,7 +29,7 @@ See :ref:`basic-auth` for more.
 How this works
 --------------
 
-These variables are handled in the :ref:`divio-cloud-sso` module.
+These variables are handled in the :ref:`Divio Cloud single-sign-on <divio-cloud-sso>` module.
```
Comment (evildmp) [inline how-to/password-protect-project.rst]: `Divio single-sign-on` (we don't use *Divio Cloud* any more)

File: how-to/resources-configure-git.rst [hunk 18]
```diff
@@ -295,7 +295,7 @@ Check that no other files are in the branch.
 You have no webhook set up
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Although the Control Panel has been able to connect to the repository and authenticate, a webhook has not yet been set
+Although the Control Panel has been able to connect to the repository and authenticates, a webhook has not yet been set
```
Comment (evildmp) [inline how-to/resources-configure-git.rst]: Please revert, original is preferred.

File: how-to/resources-pin-dependencies.rst [hunk 19]
```diff
@@ -5,18 +5,18 @@ How to pin all of your project's Python dependencies
 
 .. _pinning-dependencies-good-practice:
 
-Pinning dependencies is good practice
--------------------------------------
+Pinning dependencies is a good practice
```
Comment (evildmp) [inline how-to/resources-pin-dependencies.rst]: Please revert, original is preferred.

File: how-to/resources-pin-dependencies.rst [hunk 20]
```diff
@@ -5,18 +5,18 @@ How to pin all of your project's Python dependencies
...
+If a dependency is *unpinned* (that is, a particular version is not specified in the project's requirements), pip will
+install the latest version it finds, even if a different version was previously installed. This can cause your
+project to fail with a deployment error or worse, a runtime error, the next time it is built - *even if you didn't
 change anything in it yourself*.
 
 We strongly recommend that when you add a dependency to a project via its ``requirements.in`` that you pin it to a
 particular version, by specifying its version number.
 
-For example, if you use ``rsa`` and know that version ``1.3.4`` works, specify it::
+For example, if you use ``rsa`` and you know that version ``1.3.4`` works, specify it::
```
Comment (evildmp) [inline how-to/resources-pin-dependencies.rst]: Please revert, original is preferred.

File: how-to/resources-pin-dependencies.rst [hunk 21]
```diff
@@ -25,10 +25,9 @@ version the next time you build or redeploy it.
 
 
 What about dependencies of dependencies?
------------------------------------------------
+------------------------------------------
 
-However, dependencies can in turn themselves have dependencies. Even if you pin *your* requirements, their dependencies
-may be unpinned.
+Even if you pin *your* requirements, their dependencies may be unpinned.
```
Comment (evildmp) [inline how-to/resources-pin-dependencies.rst]: Please revert, original is preferred.

File: how-to/configure-external-logging.rst (hunk 1 above)
Comment (mebdivio) [reply how-to/configure-external-logging.rst]: Done

File: how-to/create-addon.rst (hunk 2 above)
Comment (mebdivio) [reply how-to/create-addon.rst]: Done

File: how-to/create-addon.rst (hunk 3 above)
Comment (mebdivio) [reply how-to/create-addon.rst]: Done

File: how-to/debug-dependency-conflict.rst (hunk 4 above)
Comment (mebdivio) [reply how-to/debug-dependency-conflict.rst]: Done

File: how-to/install-system-packages.rst (hunk 5 above)
Comment (mebdivio) [reply how-to/install-system-packages.rst]: Done

File: how-to/password-protect-project.rst (hunk 17 above)
Comment (mebdivio) [reply how-to/password-protect-project.rst]: Done

File: how-to/resources-configure-git.rst (hunk 18 above)
Comment (mebdivio) [reply how-to/resources-configure-git.rst]: Done

File: how-to/resources-pin-dependencies.rst (hunk 19 above)
Comment (mebdivio) [reply how-to/resources-pin-dependencies.rst]: Done

File: how-to/resources-pin-dependencies.rst (hunk 20 above)
Comment (mebdivio) [reply how-to/resources-pin-dependencies.rst]: Done

File: how-to/resources-pin-dependencies.rst (hunk 21 above)
Comment (mebdivio) [reply how-to/resources-pin-dependencies.rst]: Done

File: how-to/install-system-packages.rst (hunk 6 above)
Comment (mebdivio) [reply how-to/install-system-packages.rst]: Done

File: how-to/interact-database.rst (hunk 7 above)
Comment (mebdivio) [reply how-to/interact-database.rst]: Done

File: how-to/interact-database.rst (hunk 8 above)
Comment (mebdivio) [reply how-to/interact-database.rst]: Done

File: how-to/interact-storage.rst (hunk 9 above)
Comment (mebdivio) [reply how-to/interact-storage.rst]: Done

File: how-to/interact-storage.rst (hunk 10 above)
Comment (mebdivio) [reply how-to/interact-storage.rst]: Done

File: how-to/interact-storage.rst (hunk 11 above)
Comment (mebdivio) [reply how-to/interact-storage.rst]: Done

File: how-to/live-performance.rst (hunk 12 above)
Comment (mebdivio) [reply how-to/live-performance.rst]: Done

File: how-to/live-performance.rst (hunk 13 above)
Comment (mebdivio) [reply how-to/live-performance.rst]: Done

File: how-to/local-in-live-mode.rst (hunk 15 above)
Comment (mebdivio) [reply how-to/local-in-live-mode.rst]: Done

File: how-to/live-performance.rst (hunk 14 above)
Comment (mebdivio) [reply how-to/live-performance.rst]: Done

File: how-to/local-in-live-mode.rst (hunk 16 above)
Comment (mebdivio) [reply how-to/local-in-live-mode.rst]: Done
